### DISCUSSION TITLE
Need Help:  [Quasar] boot error: TypeError: Expected a string as value

### DISCUSSION THREAD

**Discussion by Junvary** [quasarframework/quasar#12136] [General - Components / Directives / etc] [answered]
After "quasar build", and serve with nginx（QuasarV2）
When I refresh my chrome, i can’t open my website usually with this error: [Quasar] boot error: TypeError: Expected a string as value
![QQ图片20220121090853](https://user-images.githubusercontent.com/26101389/150447268-493cd89f-c003-4c7a-88ec-97704fec3f01.png)
or open it with this warning:
![QQ图片20220121091315](https://user-images.githubusercontent.com/26101389/150447681-45d0bd0a-a794-4440-ad71-9e17af12bff0.png)

How can i fix this? Please help me...

**Comment by yusufkandemir** [comment, marked as answer]
`setCssVar` throws this error if the value(2nd parameter) isn't a string. So, double-check the value you are passing. If you can't figure it out, you can send the related boot file here and we can try to help.

About warnings, it might be coming from a UI-related dependency. In general, I would recommend building with `quasar build --debug` when you are debugging. It helps pin down the actual problem.

**Reply by Junvary** [reply]
```js
import { boot } from 'quasar/wrappers'
import { Cookies, setCssVar, getCssVar } from 'quasar'

export default boot(({ app }) => {
  const primary = Cookies.get('gqa-theme-primary') || getCssVar('primary')
  const secondary = Cookies.get('gqa-theme-secondary') || getCssVar('secondary')
  const accent = Cookies.get('gqa-theme-accent') || getCssVar('accent')
  const dark = Cookies.get('gqa-theme-dark') || getCssVar('dark')
  const positive = Cookies.get('gqa-theme-positive') || getCssVar('positive')
  const negative = Cookies.get('gqa-theme-negative') || getCssVar('negative')
  const info = Cookies.get('gqa-theme-info') || getCssVar('info')
  const warning = Cookies.get('gqa-theme-warning') || getCssVar('warning')

  setCssVar("primary", primary)
  setCssVar("secondary", secondary)
  setCssVar("accent", accent)
  setCssVar("dark", dark)
  setCssVar("positive", positive)
  setCssVar("negative", negative)
  setCssVar("info", info)
  setCssVar("warning", warning)
})
```

**Reply by yusufkandemir** [reply]
You need to debug your app to see the actual value of `primary`, `secondary`, etc. variables. Make sure they are strings.
If you still can't figure out the problem, then you need to share a repro. You can use https://codesandbox.quasar.dev (_better_) or share a public GitHub repo link.

**Reply by Junvary** [reply]
when i run "quasar dev" or "quasar serve", it's all ok.
but after "quasar build", somtimes i cant't get that actual values. (it's null)
i don't know why....
![QQ图片20220122093743](https://user-images.githubusercontent.com/26101389/150619713-8a399f28-289d-488d-986b-de678f4c0217.png)

**Reply by Junvary** [reply]
That's all I can do：
const primary = Cookies.get('gqa-theme-primary') || getCssVar('primary') || '#1976D2'
const secondary = Cookies.get('gqa-theme-secondary') || getCssVar('secondary') || '#26A69A'
const accent = Cookies.get('gqa-theme-accent') || getCssVar('accent') || '#9C27B0'
const dark = Cookies.get('gqa-theme-dark') || getCssVar('dark') || '#1D1D1D'
const positive = Cookies.get('gqa-theme-positive') || getCssVar('positive') || '#21BA45'
const negative = Cookies.get('gqa-theme-negative') || getCssVar('negative') || '#C10015'
const info = Cookies.get('gqa-theme-info') || getCssVar('info') || '#31CCEC'
const warning = Cookies.get('gqa-theme-warning') || getCssVar('warning') || '#F2C037'

Maybe it is a problem with "||" or "getCssVar()" ?

**Reply by yusufkandemir** [reply]
You probably didn't define your brand colors as shown in https://quasar.dev/style/color-palette#setting-up-defaults. You need to define them in order for `getCssVar` to return their values.

Also, when the cookies aren't set, the code is basically doing `setCssVar('primary', getCssVar('primary'))` which is redundant, creating unnecessary DOM access.

I would recommend something like this:
```js
const colorNames = [
  'primary',
  'secondary',
  'accent',
  'dark',
  'positive',
  'negative',
  'info',
  'warning',
]

for (const colorName of colorNames) {
  const value = Cookies.get(`gqa-theme-${colorName}`)

  if (value) {
    setCssVar(colorName, value)
  }
}
```
It gives the ability to add/update/remove colors easily and to change the cookie key in one place. Also, this way you also try to set the variable, only if such a value exists.
